docs/release-candidate.md

```diff
@@ -2,87 +2,35 @@
 
 ## Promote packages & artifacts
```

Comment:
Should we consolidate both release-candidate.md and release.md into one doc called promote.md? It could apply to promotion between any two channels (CI->RC or RC->stable).

Reply:
and then referenced in the release doc.

Reply:
Promotion is part of the release process but the release process is not part of promotion. Promoting to RC doesn't require any of the other pieces. I wanted to reduce the chance of confusion between the two and provide separate, straight-forward instructions.